libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3642,6 +3642,12 @@ bool TypeChecker::visit(Identifier const& _identifier)
 				_identifier.location(),
 				"\"suicide\" has been deprecated in favour of \"selfdestruct\"."
 			);
+		else if (_identifier.name() == "selfdestruct" && fType->kind() == FunctionType::Kind::Selfdestruct)
+			m_errorReporter.warning(
+				5159_error,
+				_identifier.location(),
+				"\"selfdestruct\" has been deprecated."
```

Comment:
I think this message is of little value (and may be even just outright confusing) as it stands. Should explain what deprecations means here or in the documentation (and point users to it).

Reply:
Most of the deprecation messages for the form "X has been deprecated in favour of Y". However, we don't have a Y replacement for X, so I'm not really sure how to expand the message further, other than to say it's deprecated. I can also perhaps reference the EIP in question, but further than that, no idea. Also, how is `X has been deprecated` confusing - it's standard nomenclature in compilers, and it's fairly clear what it means.

Reply:
Maybe something like this?
```
"selfdestruct" has been deprecated. The underlying opcode is likely to be removed from the EVM in the future and its use new contracts is not recommended.
```

Reply:
> Also, how is X has been deprecated confusing

Unclear what deprecated means here, because we have a number of things at hand:
a) Solidity versions (i.e. language changes)
b) EVM versions
c) Various chains

Does the deprecation apply to the language (i.e. can I still use it in assembly)? Does it apply to EVM (i.e. if I use london as a target it works, but if I use paris it doesn't)? Does it apply to various chains (i.e. Mainnet vs Gnosis Chain)?

Reply:
Alright, in that case, the only thing I'm iffy about is the `is likely to be removed` wording - can I omit this `likely` and have this be `is to be removed`?

Reply:
Probably. From how things look like now, it will likely happen but it's not like it's guaranteed. Everyone seems to want it but they might still end up not being able to agree on the specific solution that's backwards-compatible enough. Still, the fact that we're even adding this warning means that we believe it will happen so I guess something like `will eventually be removed` would be ok.

Reply:
That works for me 👍🏻